app/code/Magento/SalesRule/Model/Coupon/Massgenerator.php

```diff
@@ -245,4 +245,16 @@ public function getGeneratedCount()
     {
         return $this->generatedCount;
     }
+    
+    /**
+     * Calculate Maximum Probability based current coupon generate request
+     *
+     * @return int
+     */
+    public function getMaxProbability()
```

Comment:
What is the reason of introducing a public function without any usages?

Also, if this PR is a part of https://github.com/magento/magento2/pull/23475 fix - can you please group these changes in a single pull request

Reply:
@sivaschenko 
This method is called within the same class at `line #200` while it is not define anywhere in an interface even not getting any value because form post data is not availed there. That's why it is calling a const value aways which is defined `MAX_PROBABILITY_OF_GUESSING = 0.25;`

In the new public method, we are calculating the `max_probability` base on coupon qty and length which allow generating coupon code with expected length.

PR #23475 updated the const value of max_probability which will be enough up to two digit length as per the current scenario of issue. It can increase the digits as per the length of coupond code and it will take care by the newly defined method.

Reply:
@praful-rajput thanks for the clarification! 

I believe it would be more comfortable to process these 2 PRs together (ideally as 1 PR) considering that they are fixing the same single issue. So I suggest closing https://github.com/magento/magento2/pull/23475 and continuing work in this PR.

I have two backward compatibility concerns about changes in those 2 PRs:

- I'm not sure if constants should be changed.`max_probablity` and `max_attempts` can be provided via `data` constructor argument.
- Implementing `getMaxProbability` method breaks backward compatibility as third-party code can rely on o  providing `max_probablity` via argument and retrieving it via `getMaxProbability` method

Reply:
If #23475 seems right fixes then we would like to go with that. As per my review, both solutions will have a backward compatibility problem. This PR should be port for the older version as well. `getMaxProbability` method doesn't exist into the code since the previous version. Even it is not available from setter/getter.

Reply:
@praful-rajput `getMaxProbability` should currently return `max_probablity` from `$data` constructor argument. Introducing the new implementation for this method is definitely backwards incompatible.

Also, max probability is designed to be a setting that is influencing code length. I don't see any reason we might need to calculate max probability of the fly based on length, to use it later for the same length verification.